packages/block-editor/src/components/list-view/block.js

```diff
@@ -116,26 +138,191 @@ function ListViewBlock( {
 	const {
 		expand,
 		collapse,
+		collapseAll,
 		BlockSettingsMenu,
 		listViewInstanceId,
 		expandedState,
 		setInsertedBlock,
 		treeGridElementRef,
+		rootClientId,
 	} = useListViewContext();
+	const isMatch = useShortcutEventMatch();
+
+	// Determine which blocks to update:
+	// If the current (focused) block is part of the block selection, use the whole selection.
+	// If the focused block is not part of the block selection, only update the focused block.
+	function getBlocksToUpdate() {
+		const selectedBlockClientIds = getSelectedBlockClientIds();
+		const isUpdatingSelectedBlocks =
+			selectedBlockClientIds.includes( clientId );
+		const firstBlockClientId = isUpdatingSelectedBlocks
+			? selectedBlockClientIds[ 0 ]
+			: clientId;
+		const firstBlockRootClientId =
+			getBlockRootClientId( firstBlockClientId );
+
+		const blocksToUpdate = isUpdatingSelectedBlocks
+			? selectedBlockClientIds
+			: [ clientId ];
 
-	// If multiple blocks are selected, deselect all blocks when the user
-	// presses the escape key.
-	const onKeyDown = ( event ) => {
-		if (
-			event.keyCode === ESCAPE &&
-			! event.defaultPrevented &&
-			selectedClientIds.length > 0
-		) {
+		return {
+			blocksToUpdate,
+			firstBlockClientId,
+			firstBlockRootClientId,
+			selectedBlockClientIds,
+		};
+	}
+
+	/**
+	 * @param {KeyboardEvent} event
+	 */
+	async function onKeyDown( event ) {
+		if ( event.defaultPrevented ) {
+			return;
+		}
+
+		// If multiple blocks are selected, deselect all blocks when the user
+		// presses the escape key.
+		if ( event.keyCode === ESCAPE && selectedClientIds.length > 0 ) {
```

Comment:
This needs to be converted to a `core/block-editor/unselect` event, and there's no need to trigger an event when only one block is selected.

I'll create a follow-up PR for this.

Reply:
Here's the follow-up #61223. I left the block count check after realizing that List View allows deselecting a single block.